### DISCUSSION TITLE
OpenSSL CMS decryption not working with external providers

### DISCUSSION THREAD

**Discussion by vchandap** [openssl/openssl#25138] [Q&A] [answered]
Hi,

I am trying to use OpenSSL CMS with an external provider (https://github.com/NXPPlugNTrust/se05x-openssl-provider/tree/main).

Currently, decryption is failing due to incorrect padding. It seems that OpenSSL RSA decryption doesn't call EVP_PKEY_CTX_set_rsa_padding when PKCS1.5 padding is used and expects providers to use RSA_PKCS1_PADDING as the default.

Is the above assumption correct, and is there any documentation available for this?

Thanks, in advance

**Comment by t8m** [comment, marked as answer]
The OpenSSL default and fips providers set the PKCS#1 v1.5 padding as default for keys of type RSA and RSA-PSS padding as default for keys of type RSA-PSS. Unfortunately this does not seem to be documented.
